File: frameworks/dotnet/fastendpoints/UnitTests/ParametersTests.cs
namespace UnitTests;

public sealed class ParametersTests(App app) : TestBase<App>
{
    // rb:test parameters.static
    [Fact]
    [Trait("corpus", "parameters.static")]
    public async Task The_literal_route_wins_over_the_capture()
    {
        using HttpResponseMessage response = await app.Client.GetAsync("/parameters/static/segment/literal", Cancellation);

        await Answer.IsAsync(Expected.Json("items.small.json"), response);
    }

    // rb:test parameters.one
    [Fact]
    [Trait("corpus", "parameters.one")]
    public async Task One_capture_is_bound_as_an_integer_and_echoed()
    {
        using HttpResponseMessage response = await app.Client.GetAsync("/parameters/4821/segment/literal", Cancellation);

        await Answer.IsAsync(Expected.WithEcho("items.small.json", new() { ["one"] = 4821 }), response);
    }

    // rb:test parameters.two
    [Fact]
    [Trait("corpus", "parameters.two")]
    public async Task Two_captures_are_bound_and_echoed()
    {
        using HttpResponseMessage response = await app.Client.GetAsync("/parameters/4821/with-second/7390", Cancellation);

        await Answer.IsAsync(Expected.WithEcho("items.small.json", new() { ["one"] = 4821, ["two"] = 7390 }), response);
    }

    [Fact]
    public async Task A_capture_that_is_not_an_integer_is_refused_by_the_binder_naming_it()
    {
        using HttpResponseMessage response = await app.Client.GetAsync("/parameters/abc/segment/literal", Cancellation);

        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
        Assert.Equal(["one"], (await Answer.JsonAsync(response))["errors"]!.AsObject().Select(e => e.Key));
    }
}

File: frameworks/java/helidon-se/UnitTests/src/test/java/unittests/ParametersTests.java
package unittests;

import com.fasterxml.jackson.databind.node.ObjectNode;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

class ParametersTests extends HelidonApp {

    // rb:test parameters.static
    @Test
    @Tag("parameters.static")
    void theLiteralSegmentWinsOverACapture() throws Exception {
        Answer.is(Expected.json("items.small.json"), get("/parameters/static/segment/literal"));
    }

    // rb:test parameters.one
    @Test
    @Tag("parameters.one")
    void oneCaptureIsBoundAsAnInteger() throws Exception {
        ObjectNode echo = Expected.JSON.createObjectNode().put("one", 4821);

        Answer.is(Expected.withEcho("items.small.json", echo), get("/parameters/4821/segment/literal"));
    }

    // rb:test parameters.two
    @Test
    @Tag("parameters.two")
    void twoCapturesAreBoundAsIntegers() throws Exception {
        ObjectNode echo = Expected.JSON.createObjectNode().put("one", 4821).put("two", 7390);

        Answer.is(Expected.withEcho("items.small.json", echo), get("/parameters/4821/with-second/7390"));
    }
}
